feat: dialog polishing

marcjulian opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our
      guidelines: https://github.com/spartan-ng/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

### Primitives

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] breadcrumb
- [ ] button
- [ ] calendar
- [ ] card
- [ ] carousel
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [x] dialog
- [ ] dropdown-menu
- [ ] form-field
- [ ] hover-card
- [ ] icon
- [ ] input
- [ ] input-otp
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toggle
- [ ] toggle-group
- [ ] tooltip
- [ ] typography

### Others

- [ ] trpc
- [ ] nx
- [ ] repo
- [ ] cli

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Dialog close button height is not 16px instead 24px and focus ring is in a weird shape

<img width="36" height="40" alt="CleanShot 2025-08-19 at 10 15 49" src="https://github.com/user-attachments/assets/8a66646b-a113-4650-bc7c-081421610466" />
<img width="133" height="63" alt="CleanShot 2025-08-19 at 10 16 04" src="https://github.com/user-attachments/assets/adc692a9-3f97-47cb-9455-5e7b1540da53" />


## What is the new behavior?

- update dialog styles to the latest shadcn/ui styles and sort classes
- fix dialog close button height w/ flex classes
- add `--radius-xs: 0.125rem` to tailwind preset - is used for the dialog close button border radius in shadcn/ui (other components use it too)
- update docs to include example headings, update form field and preview
- update bottom nav in docs

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

marcjulian commented:
@goetzrobin @ashley-hunter Dialog content uses the following CSS class in the main branch, which seems to be invalid: `[animation-duration:200]`

<img width="231" height="54" alt="CleanShot 2025-08-20 at 09 03 32" src="https://github.com/user-attachments/assets/3942bece-1378-44de-bf0a-cbfd1e3d230a" />

What is it used for can we replace it with `duration-200`?

ashley-hunter commented:
> @goetzrobin @ashley-hunter Dialog content uses the following CSS class in the main branch, which seems to be invalid: `[animation-duration:200]`
> 
> <img alt="CleanShot 2025-08-20 at 09 03 32" width="231" height="54" src="https://private-user-images.githubusercontent.com/8985933/479861449-3942bece-1378-44de-bf0a-cbfd1e3d230a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU2ODA5MzEsIm5iZiI6MTc1NTY4MDYzMSwicGF0aCI6Ii84OTg1OTMzLzQ3OTg2MTQ0OS0zOTQyYmVjZS0xMzc4LTQ0ZGUtYmYwYS1jYmZkMWUzZDIzMGEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MjBUMDkwMzUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWQ1OTE4YTczMmE0ZTE5OWQ3N2UxMTRiMWJjMDE3NGUxYzk2OTA1ODdmZWY4MmEyNmRhNzAxYTM0ZjJmMDIyMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.McJxun5R8PLt11LQyxPTzEVlC_BqEDgWFqDCyUO_BBY">
> What is it used for can we replace it with `duration-200`? When using `duration-200` it is overridden by `data-[state=open]:animate-in` which sets the animation duration to 150ms

Looks like it was intentionally switched from duration-200 at one point, @goetzrobin do you recall the reason?

https://github.com/spartan-ng/spartan/commit/fe18da05740f119cf62fe17279b97255a64a66e4

marcjulian commented:
Looks like it only exists in hlm-dialog-content. The components are already updated and using `duration-200` again, maybe because of updating to the New York styles

ashley-hunter commented:
Yeah I'm sure this is fine to change back. You're probably right, when copying over the NY styles I likely replaced all the other usages without even realising.

marcjulian commented:
I updated the styles for the duration also for `hlm-alert-dialog-content`. I also updated `hlm-dialog-content` classes to be the same as with `hlm-alert-dialog-content`.

marcjulian commented:
One more thing I noticed is that the dialog is not centered on mobile in Firefox and Safari. Looks like `justify-self-center` only works currently in Chrome. Firefox and Safari expect the parent container to be `display: grid` to center it properly. However, the parent container is `cdk-dialog-container` not sure if we have access to it. 

<img width="460" height="155" alt="CleanShot 2025-08-21 at 10 16 23" src="https://github.com/user-attachments/assets/fe467a62-3cee-4574-9d87-8b161499e9e1" />

<img width="395" height="853" alt="CleanShot 2025-08-21 at 10 16 30" src="https://github.com/user-attachments/assets/14a76b68-e868-417a-bc6d-5ac1d56ab22b" />

When using `mx-auto sm:mx-0` it works fine on all three browsers.
